tiledb/sm/subarray/subarray_partitioner.cc

```diff
@@ -761,16 +769,17 @@ Status SubarrayPartitioner::compute_current_start_end(bool* found) {
   std::vector<std::vector<Subarray::MemorySize>> memory_sizes;
   RETURN_NOT_OK(subarray_.compute_relevant_fragment_est_result_sizes(
       names,
-      state_.start_,
-      state_.end_,
+      tile_overlap->range_idx_start(),
```

Comment:
I understand here why you wanted to precompute the maximum possible range. The rest of the cost is more or less intact. We could still alternatively implement what I suggested above (which will have higher accuracy in terms of the final range) in the future.